File: 0127-word-ladder/0127-word-ladder.cpp
class Solution {
public:
    int ladderLength(string beginWord, string endWord, vector<string>& wordList) {
     unordered_set<string>setList;
       setList.insert(wordList.begin(),wordList.end());
       queue<pair<string,int>>Q;
       Q.push({beginWord,1});
       setList.erase(beginWord);
       while( !Q.empty() ) {
          string word = Q.front().first;
          int level = Q.front().second;
              Q.pop();
            if(word == endWord) {
                return level ;
            }
             for(int i = 0 ; word[i] ;i++) {
                  char original = word[i];
                   for(char ch = 'a' ; ch <= 'z' ;ch++) {
                            word[i] = ch;
                       if(setList.find(word) != setList.end()){
                           Q.push({word,level+1});
                           setList.erase(word);
                       }  
                   }
                  word[i] = original; 
             }
       }
      return 0;
    }
};

File: 0127-word-ladder/0127-word-ladder.java
class Solution {
    public int ladderLength(String beginWord, String endWord, List<String> wordList) {
       HashSet<String>wordListSet = new HashSet<>(wordList);
       Queue<Pair<String,Integer>>Q = new LinkedList<>();
       Q.add(new Pair<>(beginWord,1));
       wordListSet.remove(beginWord);
       while(!Q.isEmpty()){
         Pair<String,Integer> p = Q.poll();
         String word = p.getKey();
         int level = p.getValue();
          if(word.equals(endWord)){
             return level;
          }
          char arr[] = word.toCharArray();
          for(int i = 0 ; i < arr.length;i++){
              char oChar = arr[i];
               for(char ch = 'a' ; ch <= 'z' ; ch++) {
                     arr[i] = ch;
                     String newWord = new String(arr);
                    if(wordListSet.contains(newWord)){
                        Q.add(new Pair<>(newWord,level+1));
                        wordListSet.remove(newWord);
                    }   
               }
             arr[i] = oChar;  
          } 
       }
      return 0;  
    }
}